### PR TITLE
[3] privacy request by logged in user only

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #35466 .

### Summary of Changes
privacy requests and confirmation can be made by logged-in users only 

### Testing Instructions

- Create two users, privacy1 and privacy2 with emails privacy1@exmple.com and privacy2@example.com
- Log inside the site using privacy2 user
- Create a new Privacy Request. When asked to, provide the email address privacy1@example.com instead of privacy2@example.com

### Actual result BEFORE applying this Pull Request

possible

### Expected result AFTER applying this Pull Request

not more possible you must be logged in for privacy request & confirmation

cc @tampe125 

### Documentation Changes Required
probably yes

### CODE REVIEW

File: components/com_privacy/models/request.php [hunk 1]
```diff
@@ -69,6 +68,9 @@ public function createRequest($data)
 			return false;
 		}
 
+		// Get the user email address
+		$data['email'] = JFactory::getUser()->email;
```
Comment (PhilETaylor) [inline components/com_privacy/models/request.php:72]: Please dont use deprecated code in new PR's use `Factory->getApplication()->getIdentity()`

File: components/com_privacy/models/confirm.php [hunk 2]
```diff
@@ -59,6 +58,9 @@ public function confirmRequest($data)
 			return false;
 		}
 
+		// Get the user email address
+		$data['email'] = JFactory::getUser()->email;
```
Comment (PhilETaylor) [inline components/com_privacy/models/confirm.php:62]: Please dont use deprecated code in new PR's use Factory->getApplication()->getIdentity()

File: components/com_privacy/models/request.php (hunk 1 above)
Comment (PhilETaylor) [reply components/com_privacy/models/request.php:72]: scratch that - I see this is aimed at `3.10-dev` :-)

File: components/com_privacy/controller.php [hunk 3]
```diff
@@ -30,11 +30,11 @@ public function display($cachable = false, $urlparams = array())
 	{
 		$view = $this->input->get('view', $this->default_view);
 
-		// Submitting information requests through the frontend is restricted to authenticated users at this time
-		if ($view === 'request' && JFactory::getUser()->guest)
+		// Submitting information requests and confirmation through the frontend is restricted to authenticated users at this time
+		if ((in_array($view, array('confirm', 'request')))  && JFactory::getUser()->guest)
```
Comment (Quy) [inline components/com_privacy/controller.php]:
```suggestion
		if (in_array($view, array('confirm', 'request')) && JFactory::getUser()->guest)
```

**Comment by dgrammatiko** [comment]
I have tested this item :white_check_mark: successfully on 31bd4a587986a2adfd9953c6a3c1cab29514e8be<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35470">issues.joomla.org/tracker/joomla-cms/35470</a>.</sub>

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on 31bd4a587986a2adfd9953c6a3c1cab29514e8be<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35470">issues.joomla.org/tracker/joomla-cms/35470</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35470">issues.joomla.org/tracker/joomla-cms/35470</a>.</sub>

**Comment by zero-24** [comment]
Merging here thanks :)

**Pull request merged**